clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/CloudMetricController.groovy

```diff
@@ -0,0 +1,102 @@
+import com.netflix.spinnaker.clouddriver.model.CloudMetricProvider
+import com.netflix.spinnaker.clouddriver.model.CloudMetricStatistics
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.time.Clock
+
+@RequestMapping("/cloudMetrics")
+@RestController
+class CloudMetricController {
+
+  @Autowired
+  final List<CloudMetricProvider> metricProviders
+
+  final Clock clock
+
+  @RequestMapping(method = RequestMethod.GET, value = "/{cloudProvider}/{account}/{region}")
+  List<CloudMetricDescriptor> findAll(@PathVariable String cloudProvider,
+                        @PathVariable String account,
+                        @PathVariable String region,
+                        @RequestParam Map<String, String> filters) {
+
+    getProvider(cloudProvider).listMetricDescriptors(account, region, filters)
+  }
+
+  CloudMetricController(List<CloudMetricProvider> metricProviders, Clock clock) {
+    this.clock = clock;
+    this.metricProviders = metricProviders
+  }
+
+  @Autowired
+  CloudMetricController(List<CloudMetricProvider> metricProviders) {
+    this(metricProviders, Clock.systemDefaultZone())
+  }
+
+  /**
+   * Returns a data set describing the metric over the specified period of time. If start and end times are not provided,
+   * they will be defaulted to 24 hours ago and now, respectively
+   * @param cloudProvider the provider, e.g. "aws", "gce", etc.
+   * @param account the account of the provider
+   * @param region the region applicable to the metric
+   * @param metricName the name of the metric
+   * @param startTime start time (inclusive)
+   * @param endTime end time (exclusive)
+   * @param filters any provider-specific filters used to create the data set
+   * @return the set of statistics
+   */
+  @RequestMapping(method = RequestMethod.GET, value = "/{cloudProvider}/{account}/{region}/{metricName}/statistics")
+  CloudMetricStatistics getStatistics(@PathVariable String cloudProvider,
+                                      @PathVariable String account,
+                                      @PathVariable String region,
+                                      @PathVariable String metricName,
+                                      @RequestParam(required = false) Long startTime,
+                                      @RequestParam(required = false) Long endTime,
+                                      @RequestParam Map<String, String> filters) {
+
+    // Spring will map all request parameters into the filters map, so it's best to remove the declared ones
+    filters.remove("startTime")
+    filters.remove("endTime")
+
+    Long start = startTime ?: clock.instant().minus(24, HOURS).toEpochMilli()
```

Comment:
The only downside to the relative window is: how do I get a non-relative window, e.g. if I have today's data but also want to see the day before, or if I want to see what Friday nights look like over the past month.

We could add a parameter (I can't think of a name for it, maybe `relativeLookbackInMs` or something along those lines), but I think we'd still want the option for start/end times to be specified.

And we'll have to strip any of these parameters out of the `filters`, just because Spring is going to put all the request parameters into that map, even if they're specified as `@RequestParam`s in the method signature.

For now, I'm going to leave this as-is, but, again, am open to refactoring when you get to the Google implementation.